code/modules/mob/living/carbon/human/species.dm

```diff
@@ -1411,9 +1411,84 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 	if(user == target)
 		return FALSE
 	if(user.loc == target.loc)
+		if(user.pulling == target && user.zone_selected == BODY_ZONE_HEAD && user.grab_state >= GRAB_NECK)
+			start_noogie(user, target)
 		return FALSE
 	user.disarm(target)
 
+/// For starting a noogie, leads into the actual noogie loop
+/datum/species/proc/start_noogie(mob/living/carbon/human/user, mob/living/carbon/human/target)
+	if(!(target?.get_bodypart(BODY_ZONE_HEAD)) || user.pulling != target || user.getStaminaLoss() > 80)
+		return FALSE
+
+	var/obj/item/bodypart/head/the_head = target.get_bodypart(BODY_ZONE_HEAD)
+	if(!(HAS_FLESH in target.dna?.species.species_traits) || !the_head.is_organic_limb())
+		to_chat(user, "<span class='warning'>You can't noogie [target], [target.p_they()] have no skin on [target.p_their()] head!</span>")
+		return
+
+	// [user] gives [target] a [prefix_desc] noogie[affix_desc]!
+	var/prefix_desc = "rough"
+	var/affix_desc = ""
+	var/affix_desc_target = ""
+
+	if(HAS_TRAIT(target, TRAIT_ANTENNAE))
+		prefix_desc = "violent"
+		affix_desc = "on [target.p_their()] sensitive antennae"
+		affix_desc_target = "on your highly sensitive antennae"
+	if(user.dna?.check_mutation(HULK))
+		prefix_desc = "sickeningly brutal"
+
+	var/message_others = "[prefix_desc] noogie[affix_desc]"
+	var/message_target = "[prefix_desc] noogie[affix_desc_target]"
+
+	user.visible_message("<span class='danger'>[user] begins giving [target] a [message_others]!</span>", "<span class='warning'>You start giving [target] a [message_others]!</span>", vision_distance=COMBAT_MESSAGE_RANGE, ignored_mobs=target)
+	to_chat(target, "<span class='userdanger'>[user] starts giving you a [message_target]!</span>")
+
+	if(!do_after(user, 1.5 SECONDS, target))
+		to_chat(user, "<span class='warning'>You fail to give [target] a noogie!</span>")
+		to_chat(target, "<span class='danger'>[user] fails to give you a noogie!</span>")
+		return
+
+	if(prefix_desc == "rough")
```

Comment:
I'd rather this just be a true/false variable, rather than relying on the specific string directly.